src/base/bittorrent/session.cpp

```diff
@@ -1117,13 +1120,16 @@ void Session::prepareStartup()
     connect(this, &Session::torrentsLoaded, context, [this, context](const QVector<Torrent *> &torrents)
     {
         context->processingResumeDataCount -= torrents.count();
+        context->finishedResumeDataCount += torrents.count();
         if (!context->isLoadedResumeDataHandlingEnqueued)
         {
             QMetaObject::invokeMethod(this, [this, context]() { handleLoadedResumeData(context); }, Qt::QueuedConnection);
             context->isLoadedResumeDataHandlingEnqueued = true;
         }
         m_nativeSession->post_torrent_updates();
         m_refreshEnqueued = true;
+
+        emit startupProgressUpdated((context->finishedResumeDataCount * 100) / context->totalResumeDataCount);
```

Comment:
I would calculate in floating point. Performance isn't worth for risking overflow IMO.

Reply:
> Performance isn't worth for risking overflow IMO.

Overflow is hardly achievable in practice. This is only possible if there are more torrents than (max int64 / 100). This is hardly possible. So your proposal is akin to an offer to live in a bunker for all of your life because of the possible threat of a nuclear strike.
Are you still going to insist on this? Of course, it's not a big deal to make the startup acceleration a little slower...

Reply:
If you really don't want floating point then you should put a `Q_ASSERT(context->finishedResumeDataCount  <= (std::numeric_limits<int64_t>::max() / 100))` to show the allowed/intended maximum size.

>Of course, it's not a big deal to make the startup acceleration a little slower...

You always talked about it but what is the reality?

https://stackoverflow.com/a/41570985
https://stackoverflow.com/a/36397764
https://stackoverflow.com/questions/55832817/why-float-division-is-faster-than-integer-division-in-c
Didn't found data on ARM architecture... but I assume that if the CPU is powerful enough to run qbt then there is high chance to include an FPU.

Of course, (in general) keeping calculations to integer is a good practice but it is only acceptable when there are no safety concerns in the edge cases, otherwise I won't hesitate to use floating point.

Reply:
> You always talked about it but what is the reality?

Yes, you're right. Division is the weak side of integer calculations (unless it is division by a power of two).

Reply:
> I would calculate in floating point.

Done.